File: Advanced/BotNavGuiDemo/app/src/main/java/edu/cs4730/botnavguidemo/BottomNavigationDialogFragment.java
package edu.cs4730.botnavguidemo;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.google.android.material.navigation.NavigationView;

import androidx.annotation.NonNull;

import edu.cs4730.botnavguidemo.databinding.FragmentBottomsheetBinding;

public class BottomNavigationDialogFragment extends BottomSheetDialogFragment {

    private OnFragmentInteractionListener mListener;
    private FragmentBottomsheetBinding binding;

    public interface OnFragmentInteractionListener {
        void onFragmentPicker(int id);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentBottomsheetBinding.inflate(inflater, container, false);
        binding.navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                mListener.onFragmentPicker(menuItem.getItemId());
                dismiss();
                return false;
            }
        });

        return binding.getRoot();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        Activity activity = requireActivity();
        try {
            mListener = (OnFragmentInteractionListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

}

File: Advanced/BotNavGuiDemo_kt/app/src/main/java/edu/cs4730/botnavguidemo_kt/BottomNavigationDialogFragment.kt
package edu.cs4730.botnavguidemo_kt

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import edu.cs4730.botnavguidemo_kt.databinding.FragmentBottomsheetBinding

class BottomNavigationDialogFragment : BottomSheetDialogFragment() {
    private var mListener: OnFragmentInteractionListener? = null

    lateinit var  binding: FragmentBottomsheetBinding

    interface OnFragmentInteractionListener {
        fun onFragmentPicker(id: Int)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentBottomsheetBinding.inflate(inflater, container, false)
        binding.navigationView.setNavigationItemSelectedListener { menuItem ->
            mListener!!.onFragmentPicker(menuItem.itemId)
            dismiss()
            false
        }
        return binding.root
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        val activity: Activity = requireActivity()
        mListener = try {
            activity as OnFragmentInteractionListener?
        } catch (e: ClassCastException) {
            throw ClassCastException(
                activity.toString()
                        + " must implement OnFragmentInteractionListener"
            )
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }
}
